### PR TITLE
Switch to QTableView model-based for shortcuts tab

### PR DESCRIPTION
**Pull request by CoBC**
@bear101 There is a strange issue, when double clicking on an item, keycompdlg appears and disappears immediately, if I add a QMessageBox before opening keycompdlg, all works fine.
Do you have any idea?

### CODE REVIEW

**Comment by CoBC** [comment]
Note: the issue described in the PR description is due to the release of key space (key release 32) after keycompdlg was opened.
The only way I cna find to fix it is to add a QTimer of E.G. 50ms before opening keycompdlg, but it seems to be an hack more than something else.

**Comment by bear101** [comment]
@CoBC Here's how I fixed the initial-dialog-close issue on macOS: e68b3cd4b96ab5ba39ba99d88be42dc9240018ee

The issue seems to be the same on Windows.

**Comment by CoBC** [comment]
Yes, issue is same on all platforms

**Comment by bear101** [comment]
The issue should be fixed now

**Comment by CoBC** [comment]
Yes, it works

**Comment by bear101** [comment]
@CoBC if I enable push-to-talk key in General-tab then it's not updated in "Shortcuts"-tab and vice versa.

**Comment by CoBC** [comment]
@bear101 Should we remove HOTKEY_PUSHTOTALK from shortcuts model, as it was not present here previously, or fix the issue and let this one in model also?

**Comment by amirmahdifard** [comment]
in my opinion keep push to talk in both places for eas of access, and fix the updating problem.

**Comment by bear101** [comment]
I think we should fix the issues. Push-to-talk key is a Shortcut that works in the same way as the other shortcuts. I know this becomes a bit more complicated to maintain state in Preferences dialog but it's more logical.

**Comment by CoBC** [comment]
I think I've fixed the issue

**Comment by bear101** [comment]
I'll try and fix the conflicts

**Comment by bear101** [comment]
I've fixed the conflicts and rebased with master.

File: Client/qtTeamTalk/preferencesdlg.cpp [hunk 1]
```diff
@@ -1005,35 +952,31 @@ void PreferencesDlg::slotSaveChanges()
...
+        while (ite != m_hotkeys.end())
         {
             saveHotKeySettings(ite.key(), *ite);
 #ifdef Q_OS_WIN32
             TT_HotKey_Register(ttInst, ite.key(), &(*ite)[0], INT32(ite->size()));
 #endif
             ite++;
         }
+        Hotkeys activeHotkeys = 0;
+        for (auto it = m_shortcutsmodel->getShortcuts().begin(); it != m_shortcutsmodel->getShortcuts().end(); ++it)
```
Comment (bear101) [inline Client/qtTeamTalk/preferencesdlg.cpp:974]: This code is crashing. You get a copy of shortcuts in `.begin()` and another copy of shortcuts in` .end()`. This means that iterators will never match.` getShortcuts(`) has to return a const reference:

`const hotkeys_t& ShortcutsModel::getShortcuts()`

**Comment by bear101** [comment]
@CoBC I think colon should be part of the translation string. I'm not sure colon makes sense in Chinese or other non Latin languages: [1af4e06](https://github.com/BearWare/TeamTalk5/pull/2387/commits/1af4e06ce9d33b063eb6366f08326faf40b86811)

**Comment by bear101** [comment]
Hm, I'm seeing some strange behavior:

1. Start Prefences dialog
2. Check "push to talk" in General-tab
3. Setup a hotkey.
4. Click Ok.
5. Start Preferences dialog again.
6. Now "push to talk" is unchecked.

**Comment by CoBC** [comment]
OK for colon in translation strings. Note however that we also have some 
similar strings in other dialogs like server properties, server stats, 
or user info, I didn't include colon in translation when migrating from 
QLineEdit to QLabel to avoid to retranslate strings just to add a colon. 
Should we update all translation strings to include colon?

**Comment by CoBC** [comment]
This strange behaviour should be fixes by 54e3902d9

**Comment by bear101** [comment]
Is there a reason to keep SETTINGS_GENERAL_PUSHTOTALK after introduction of SETTINGS_SHORTCUTS_ACTIVEHKS?

**Comment by CoBC** [comment]
Hmmm, right, probably not. I'll remove it

**Comment by bear101** [comment]
I've made the Key Combination dialog show up which is being configured.

**Comment by bear101** [comment]
I'm still seeing some strange behavior.

1. Set PTT to F5 in General tab
2. Press OK
3. Go back to General tab
4. Change to F6
5. Press OK
6. Now PTT is still listed as F5 in the status bar

**Comment by CoBC** [comment]
Hmm, strange, I don't reproduce it here :(

**Comment by bear101** [comment]
I've fix part of the problem here: [a8d217f](https://github.com/BearWare/TeamTalk5/pull/2387/commits/a8d217fc9c0a00940e1b1779444b6e322d3b3027)

Note the description of std::map::insert() is:
`Inserts element(s) into the container, if the container doesn't already contain an element with an equivalent key.`

**Comment by bear101** [comment]
When the check state is toggled on PTT:
`ui.pttChkBox->setChecked(true);`
Then it triggers an event that reloads PTT:
```
        hotkey_t ptt;
        if (loadHotKeySettings(HOTKEY_PUSHTOTALK, ptt))
            m_hotkeys[HOTKEY_PUSHTOTALK] = ptt;
```

Therefore PTT is always reverted.

**Comment by CoBC** [comment]
Hmmmm, I can't reproduce this one also :(

I tried enabled PTT in general tab, it's enabled correctly, I tried 
disabled in general tab, it's disabled, I tried to uncheck the PTT 
shortcut in shortcuts tab, it appears uncheck and general tab and is 
disabled, and I tried check again in shortcuts tab, it appears check in 
general tab and it's enabled.

So, not sure to understand the issue :(

**Comment by bear101** [comment]
It is the key combination that does not match

**Pull request merged**